apollo-execution-processor/src/main/kotlin/com/apollographql/execution/processor/ApolloProcessor.kt

```diff
@@ -53,6 +53,13 @@ class ApolloProcessor(
     done = true
 
     val query = getRootSymbol(resolver, KotlinSymbols.GraphQLQuery.canonicalName)
+
+    val ignoreQuery = getRootSymbol(resolver, KotlinSymbols.GraphQLIgnore.canonicalName)
+
+    if (ignoreQuery != null) {
```

Comment:
Do we need something like this? Or is it better if it is more specific?

Reply:
`getRootSymbol()` is used for the Query/Mutation/Subscription roots, these classes can't be ignored by definition, this is the entry point of the traversal algorithm. 

I think we'll want to check `@GraphQLIgnore` in 2 places:

* supertypes: https://github.com/apollographql/apollo-kotlin-execution/blob/6541b551131b031618a9c70d75f4c6568341b022/apollo-execution-processor/src/main/kotlin/com/apollographql/execution/processor/definitions.kt#L506
* fields: https://github.com/apollographql/apollo-kotlin-execution/blob/6541b551131b031618a9c70d75f4c6568341b022/apollo-execution-processor/src/main/kotlin/com/apollographql/execution/processor/definitions.kt#L384

There may be others (arguments, enumValues) maybe but those 2 would be a great start.

Reply:
Ah okay! Does that also mean that we should allow interfaces to not be sealed without the `@GraphQLIgnore` annotation in the definition? :)

Reply:
Yes. Basically any `@GraphQLIgnore` interface shouldn't be visited so it's not important whether it's generic or sealed, just do not take it into account.

Reply:
Okay, then I can remove that check I think? :)

Reply:
I think so.